### PR TITLE
Test for piping htsjdk data

### PR DESCRIPTION
**Pull request by tlangs**
### Description
A recent change in htsjdk broke the ability of pipe BAM output from stdin into a Picard command line tool. This PR adds a test to make sure that piping into picard is tested.
----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/test/java/picard/sam/PipedDataTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import picard.cmdline.CommandLineProgramTest;
+
+public class PipedDataTest extends CommandLineProgramTest {
+
+    private String classPath = "\"" + System.getProperty("java.class.path") + "\" ";
+
+
+    @Override
+    public String getCommandLineProgramName() {
+        return SortSam.class.getSimpleName();
```
Comment (magicDGS) [inline src/test/java/picard/sam/PipedDataTest.java]: I guess that this is wrong...maybe an exception saying that "this shouldn't be called" is better than assigning a name for a random tool...

Comment (danxmoran) [reply src/test/java/picard/sam/PipedDataTest.java]: Do we need to extend `CommandLineProgramTest` at all? It doesn't seem a good fit if it forces picking one tool name (but I'm not sure of all the magic it provides).

Comment (magicDGS) [reply src/test/java/picard/sam/PipedDataTest.java]: I think that in this case the "magic" (making command line with arguments and the tool name, or running the tool using `Main`) is not used anywhere, so probably is better to don't extend...

Comment (tlangs) [reply src/test/java/picard/sam/PipedDataTest.java]: This is a remnant of an older revision that I forgot to delete.

**Comment by danxmoran** [comment]
@tlangs I took another look and you were right that #1084 seems like a separate, more complicated error than just piping a bam. Can you remove the `RevertSam` test so we can merge this? HTSJDK [fixed the issue](https://github.com/samtools/htsjdk/commit/2b8919b50479c1046cae80be6dd766b3db7424eb) in 2.14.3 (which we're using now) so I'd expect the simple `SortSam` test to pass.

**Comment by kbergin** [comment]
To clean up this repo, we are deleting stale branches that have been merged or closed. You have a short period of time to restore this branch if needed. Thanks!

**Pull request merged**